test/cmdlineTests/yul_source_locations_in_asm/input.json

```diff
@@ -0,0 +1,22 @@
+{
+	"language": "Solidity",
+	"sources":
+	{
+		"C":
+		{
+			"content": "//SPDX-License-Identifier: GPL-2.0\npragma solidity >=0.0;\npragma abicoder v2;\n\ncontract C\n{\n	int constant constVar = 41;\n	int immutable immutVar = 42;\n	int public stateVar;\n\n	constructor(int _init)\n	{\n		stateVar = _init;\n	}\n\n	function f() external pure returns (int)\n	{\n		return constVar + immutVar;\n	}\n	modifier m()\n	{\n		stateVar++;\n		_;\n	}\n	function f2() m public returns (int)\n	{\n		return stateVar + this.f() + immutVar;\n	}\n}\n"
```

Comment:
The output assembly is gigantic - can we slim down the source code a bit maybe? Also please try to generate some cross-source code, maybe a free function in one file that is used from the contract in the other file.

Reply:
Maybe activating the optimizer would also help